res/controllers/Pioneer-DDJ-SX-scripts.js

```diff
@@ -0,0 +1,2149 @@
+    if (!trackLoaded) {
+        return;
+    }
+
+    if (value && !cue) {
+        // play illumination:
+        PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.playPauseDeck, true);
+        PioneerDDJSX.playFromCuePoint = true;
+        return;
+    }
+    if (!value && !cue) {
+        // pause illumination:
+        PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.playPauseDeck, false);
+        PioneerDDJSX.playFromCuePoint = false;
+        return;
+    }
+    if (value && cue) {
+        // stop and go to cue point illumination:
+        PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.cueDeck, true);
+        PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.playPauseDeck, true);
+        PioneerDDJSX.cuePressed[deck] = false;
+        PioneerDDJSX.playFromCuePoint = true;
+    }
+    if (!value && cue) {
+        if (PioneerDDJSX.playFromCuePoint) {
+            // play from cue point illumination:
+            PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.cueDeck, false);
+            PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.cueDeck, true);
+            PioneerDDJSX.playFromCuePoint = false;
+        } else {
+            // set cue point illumination:
+            PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.cueDeck, false);
+            PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.cueDeck, true);
+            PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.cueDeck, false);
+        }
+        PioneerDDJSX.cuePressed[deck] = false;
+    }
+};
+
+PioneerDDJSX.playStutterIllumination = function(value, group, control) {
+    if (value) {
+        PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.cueDeck, false);
+        PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.cueDeck, true);
+    } else {
+        PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.playPauseDeck, true);
+    }
+};
+
+PioneerDDJSX.cueLed = function(value, group, control) {
+    PioneerDDJSX.nonPadLedControl(group, PioneerDDJSX.nonPadLeds.cue, value);
+    PioneerDDJSX.nonPadLedControl(group, PioneerDDJSX.nonPadLeds.shiftCue, value);
+};
+
+PioneerDDJSX.loadLed = function(value, group, control) {
+    var deck = PioneerDDJSX.channelGroups[group];
+    PioneerDDJSX.generalLedControl(PioneerDDJSX.nonPadLeds["loadDeck" + (deck + 1)], value);
+    PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.loadedDeck, value);
+};
+
+PioneerDDJSX.reverseLed = function(value, group, control) {
+    PioneerDDJSX.nonPadLedControl(group, PioneerDDJSX.nonPadLeds.censor, value);
+    PioneerDDJSX.nonPadLedControl(group, PioneerDD
```

Comment:
I see in [Pioneer's demo video](https://www.youtube.com/watch?v=zmWahVHJWTE) that the LEDs can rotate backwards with Serato. [Another video](https://www.youtube.com/watch?v=zmWahVHJWTE) shows it working with Traktor. So it is possible, they just have not documented it. You'll probably have to intercept the MIDI messages Serato sends to the controller with [MIDI OX](http://www.midiox.com/) (Windows) or [MIDI Monitor](http://www.snoize.com/MIDIMonitor/) (Mac) to figure out what to send. You might have to turn off the setting to use the controller with DJ software other than Serato.